main.tf

```diff
@@ -80,6 +80,40 @@ resource "google_service_account" "org_terraform" {
 /***********************************************
   GCS Bucket - Terraform State
  ***********************************************/
+data "google_storage_project_service_account" "gcs_account" {
+  project = module.seed_project.project_id
+}
+
+resource "google_kms_key_ring" "key_ring" {
+  count    = var.encrypt_gcs_bucket_tfstate ? 1 : 0
+  name     = "${var.project_prefix}-keyring"
+  project  = module.seed_project.project_id
+  location = var.default_region
+}
+
+resource "google_kms_crypto_key" "gcs_key" {
+  count           = var.encrypt_gcs_bucket_tfstate ? 1 : 0
+  name            = "${var.project_prefix}-key"
+  key_ring        = google_kms_key_ring.key_ring[0].self_link
+  rotation_period = var.key_rotation_period
+
+  lifecycle {
+    prevent_destroy = false
+  }
+
+  version_template {
+    algorithm        = "GOOGLE_SYMMETRIC_ENCRYPTION"
+    protection_level = var.key_protection_level
+  }
+
+}
+
+resource "google_kms_crypto_key_iam_member" "gcs_key_iam" {
+  count         = var.encrypt_gcs_bucket_tfstate ? 1 : 0
+  crypto_key_id = google_kms_crypto_key.gcs_key[0].id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  member        = "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"
+}
```

Comment:
Could we use https://github.com/terraform-google-modules/terraform-google-kms?
```tf
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "~> 1.2"

  project_id         = module.seed_project.project_id
  location           = var.default_region
  keyring            = "${var.project_prefix}-keyring"
  keys               = [ "${var.project_prefix}-key"]
  key_rotation_period = var.key_rotation_period
  key_protection_level = var. key_protection_level
  set_decrypters_for  = ["${var.project_prefix}-key"]
  set_encrypters_for = ["${var.project_prefix}-key"]
  decrypters = [
   "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}",
  ]
  encrypters = [
   "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}",
  ]
}
```

Reply:
Definitely. Leme update.